Documentation/devel/inspect-containers.md

```diff
@@ -0,0 +1,354 @@
+...
+5710  mount("stage1/rootfs/sys/fs/cgroup/freezer", "stage1/rootfs/sys/fs/cgroup/freezer", 0xc42027d2b8, MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND, NULL) = 0
+```
+
+Now is the time to start systemd-nspawn to create the container itself:
+
+```
+5710  execve("stage1/rootfs/usr/lib/ld-linux-x86-64.so.2", ["stage1/rootfs/usr/lib/ld-linux-x86-64.so.2", "stage1/rootfs/usr/bin/systemd-nspawn", "--boot", "--notify-ready=yes", "--register=true", "--link-journal=try-guest", "--quiet", "--uuid=d5513d49-d14f-45d1-944b-39437798ddda", "--machine=rkt-d5513d49-d14f-45d1-944b-39437798ddda", "--directory=stage1/rootfs", "--capability=CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FSETID,CAP_FOWNER,CAP_KILL,CAP_MKNOD,CAP_NET_RAW,CAP_NET_BIND_SERVICE,CAP_SETUID,CAP_SETGID,CAP_SETPCAP,CAP_SETFCAP,CAP_SYS_CHROOT", "--", "--default-standard-output=tty", "--log-target=null", "--show-status=0"], 0xc4202bc0f0 /* 29 vars */ <unfinished ...>
+```
+
+Note we don't need to pass the `--private-network` option because rkt already created and configured the network using CNI.
+
+Some interesting things systemd-nspawn does are moving the container filesystem tree to `/`:
+
+```
+5747  mount("/var/lib/rkt/pods/run/d5513d49-d14f-45d1-944b-39437798ddda/stage1/rootfs", "/", NULL, MS_MOVE, NULL) = 0
+5747  chroot(".")                       = 0
+```
+
+And creating all the other namespaces: mount, UTS, IPC, and PID.
+Check [namespaces(7)][man-namespaces] for more information.
+
+```
+5747  clone(child_stack=NULL, flags=CLONE_NEWNS|CLONE_NEWUTS|CLONE_NEWIPC|CLONE_NEWPID|SIGCHLD) = 5748
+```
+
+Once it's done creating the container, it will execute the init process, which is systemd:
+
+```
+5748  execve("/usr/lib/systemd/systemd", ["/usr/lib/systemd/systemd", "--default-standard-output=tty", "--log-target=null", "--show-status=0"], 0x7f904604f250 /* 8 vars */) = 0
+```
+
+Which then will execute systemd-journald to handle logging:
+
+```
+5749  execve("/usr/lib/systemd/systemd-journald", ["/usr/lib/systemd/systemd-journald"], 0x5579c5d79d50 /* 8 vars */ <unfinished ...>
+...
+```
+
+And at some point it will execute our application's service (in this example, etcd).
+
+But first, it needs to execute its companion `prepare-app` dependency:
+
+```
+5751  execve("/prepare-app", ["/prepare-app", "/opt/stage2/etcd/rootfs"], 0x5579c5d7d580 /* 7 vars */) = 0
+```
+
+`prepare-app` bind-mounts a lot of files from stage1 to stage2, so our app has access to a reasonable environment:
+
+```
+5751  mount("/dev/null", "/opt/stage2/etcd/rootfs/dev/null", 0x49006f, MS_BIND, NULL) = 0
+5751  mount("/dev/zero", "/opt/stage2/etcd/rootfs/dev/zero", 0x49006f, MS_BIND, NULL) = 0
```

Comment:
you can refer to the spec where the list of files is defined:

https://github.com/appc/spec/blob/3f41f3e4b672d740707521cde8a5dd4626c94f0a/spec/OS-SPEC.md